src/ruby/puppetserver-lib/puppet/server/compiler.rb

```diff
@@ -14,11 +15,55 @@ def compile(request_data)
         node = create_node(processed_hash)
 
         catalog = Puppet::Parser::Compiler.compile(node, processed_hash['job_id'])
+
+        maybe_save(processed_hash, node.facts, catalog)
```

Comment:
Does this include trusted facts? Should it?

Reply:
I don't believe so. PDB gets trusted facts by looking up the "local" trusted data for the node attached to the request. 

https://github.com/puppetlabs/puppetdb/blob/master/puppet/lib/puppet/indirector/facts/puppetdb.rb#L12-L28

Not saying that I have much faith in this implementation, but it looks right to me reading that code.

Reply:
Hmm... I wonder if normally `:trusted_information` is set to something in the context. We might need to look into that.

Reply:
So turns out PDB overwrites whatever is in the facts object being saved https://github.com/puppetlabs/puppetdb/blob/master/puppet/lib/puppet/indirector/facts/puppetdb.rb#L28, by either generating a default or pulling data out of the context https://github.com/puppetlabs/puppetdb/blob/master/puppet/lib/puppet/indirector/facts/puppetdb.rb#L13-L14 Neither of which is the trusted facts we’re sending along. Given that, we may have to use the context? :/

Reply:
This is relevant for the ACE compilations, where both facts and trusted facts are supplied (rather than fetched from PDB) and expected to be saved.